lib/common/grpc-service.js

```diff
@@ -457,11 +458,13 @@ GrpcService.createDeadline_ = function(timeout) {
 GrpcService.getError_ = function(err) {
   if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
     var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
-  }
+    var errorObj = is.error(err) ? new Error() : {};
```

Comment:
The thing that confuses this for me is a line line:

```js
respError = GrpcService.getStatus_(err);
// ^-- assign an "error" from a method that gets me a "status" from an "error"
```

Given this naming pattern, I think this might be more clear:

```js
var respStatus = GrpcService.getStatus_(err);

if (is.error(respStatus)) {
  respError = respStatus;
}
```

However, can you think of another way to handle this duality? Maybe two methods; `decorateGrpcStatus_` and `decorateGrpcError_` or some kind of utility like `isGrpcError_`?